src/Behat/Behat/Definition/Pattern/Pattern.php

```diff
@@ -17,59 +17,34 @@
  */
 final class Pattern
 {
-    /**
-     * @var string
-     */
-    private $canonicalText;
-    /**
-     * @var string
-     */
-    private $pattern;
-    /**
-     * @var int
-     */
-    private $placeholderCount;
+    public function __construct(
+        private string $suggestedMethodName,
+        private string $pattern,
+        private int $placeholderCount = 0,
+    ) {
+    }
 
     /**
-     * Initializes pattern.
+     * Returns canonical step text.
      *
-     * @param string $canonicalText
-     * @param string $pattern
-     * @param int    $placeholderCount
+     * @deprecated see getSuggestedMethodName
      */
-    public function __construct($canonicalText, $pattern, $placeholderCount = 0)
+    public function getCanonicalText(): string
     {
-        $this->canonicalText = $canonicalText;
-        $this->pattern = $pattern;
-        $this->placeholderCount = $placeholderCount;
+        return $this->suggestedMethodName;
     }
 
-    /**
-     * Returns canonical step text.
-     *
-     * @return string
-     */
-    public function getCanonicalText()
+    public function getSuggestedMethodName(): string
```

Comment:
The unified diff looks at a glance like I've just renamed the method - in fact `getCanonicalText` is still present but deprecated, and `getSuggestedMethodName` is a new method.